packages/domain/lib/usecases/date_validation_usecase.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:domain/exceptions/date_validation_exception.dart';
+import 'package:domain/usecases/usecase.dart';
+
+class DateValidationUseCase implements UseCaseParams<String, void> {
+  @override
+  void call(String params) {
```

Comment:
pass DateModel instead of the String

DateModel(int? month, int? year)

let's remove parsing from the usecase, create separate mapper in bloc

Reply:
Fixed.